src/write/elf/object.rs

```diff
@@ -166,280 +170,175 @@ impl<'a> Object<'a> {
             return Ok(());
         };
 
+        let unsupported_reloc = || Err(Error(format!("unimplemented relocation {:?}", reloc)));
+
+        #[rustfmt::skip]
         let r_type = match self.architecture {
-            Architecture::Aarch64 => match (kind, encoding, size) {
-                (RelocationKind::Absolute, RelocationEncoding::Generic, 64) => elf::R_AARCH64_ABS64,
-                (RelocationKind::Absolute, RelocationEncoding::Generic, 32) => elf::R_AARCH64_ABS32,
-                (RelocationKind::Absolute, RelocationEncoding::Generic, 16) => elf::R_AARCH64_ABS16,
-                (RelocationKind::Relative, RelocationEncoding::Generic, 64) => {
-                    elf::R_AARCH64_PREL64
-                }
-                (RelocationKind::Relative, RelocationEncoding::Generic, 32) => {
-                    elf::R_AARCH64_PREL32
-                }
-                (RelocationKind::Relative, RelocationEncoding::Generic, 16) => {
-                    elf::R_AARCH64_PREL16
-                }
-                (RelocationKind::Relative, RelocationEncoding::AArch64Call, 26)
-                | (RelocationKind::PltRelative, RelocationEncoding::AArch64Call, 26) => {
-                    elf::R_AARCH64_CALL26
-                }
-                _ => {
-                    return Err(Error(format!("unimplemented relocation {:?}", reloc)));
-                }
+            A::Aarch64 => match (size, kind, encoding) {
```

Comment:
Once these are changed back to using rustfmt, I think `size` last will still be better.